### ISSUE TITLE
Bug: Insufficient GPU tensor cleanup could cause memory leaks

### ISSUE THREAD

**Issue by filthyrake** [filthyrake/telescope_cam_detection#185]
## Description
While `snapshot_saver.py:119-123` explicitly deletes GPU tensors, similar patterns aren't consistently applied elsewhere. GPU tensors should be explicitly freed after conversion to NumPy to prevent VRAM leaks.

## Impact
- High severity
- VRAM exhaustion over time
- OOM errors in long-running deployments
- Performance degradation

## Locations
- Various modules converting tensors to NumPy
- Inference pipeline
- Stage 2 processing

## Suggested Solution
1. Audit all tensor-to-numpy conversions
2. Add explicit `.cpu().detach()` before conversion
3. Call `del tensor` after use
4. Use context managers for tensor lifecycle
5. Add VRAM monitoring to health checks

## Example Pattern
```python
# Bad
numpy_array = tensor.cpu().numpy()

# Good
numpy_array = tensor.cpu().detach().numpy()
del tensor  # Explicit cleanup
torch.cuda.empty_cache()  # Aggressive cleanup
```

## Related
- Issue #98 (GPU memory leak in snapshot frame buffer) - CLOSED
- Issue #125 (Graceful degradation for GPU OOM) - CLOSED
